kotlin/services/cognito/src/main/kotlin/com/kotlin/cognito/CognitoMVP.kt

```diff
@@ -108,14 +107,34 @@ suspend fun main(args: Array<String>) {
     verifyTOTP(newSession, myCode)
     println("*** Re-enter a 6-digit code displayed in Google Authenticator")
     val mfaCode: String = inOb.nextLine()
-    val authResponse1 = initiateAuth(clientId, userName, password)
+    val authResponse1 = checkAuthMethod(clientId, userName, password, poolId)
     val session2 = authResponse1.session
     adminRespondToAuthChallenge(userName, clientId, mfaCode, session2)
 }
 
+// snippet-start:[cognito.kotlin.initiateauth.main]
+suspend fun checkAuthMethod(clientIdVal: String, userNameVal: String, passwordVal: String, userPoolIdVal: String): AdminInitiateAuthResponse {

```

Comment:
I see where it is now using the Admin method for InitiateAuth, but I don't see it for AdminRespondToAuthChallenge. Are you sure this Kotlin example is using the correct action for that?